metrics/exporter.go

```diff
@@ -177,9 +189,9 @@ func isNewExporterRequired(newConfig *metricsConfig) bool {
 }
 
 // newMetricsExporter gets a metrics exporter based on the config.
-// This function is not implicitly thread-safe.
+// This function must be called with the metricsMux reader (or writer) locked.
 func newMetricsExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
-	ce := getCurMetricsExporter()
+	ce := curMetricsExporter
```

Comment:
`ce` is only used once in this func. We can just use `curMetricsExporter` directly.